src/meta/src/hummock/compaction/picker/emergency_compaction_picker.rs

```diff
@@ -50,20 +51,62 @@ impl EmergencyCompactionPicker {
         stats: &mut LocalPickerStatistic,
     ) -> Option<CompactionInput> {
         let unused_validator = Arc::new(CompactionTaskValidator::unused());
-
-        let mut base_level_compaction_picker = LevelCompactionPicker::new_with_validator(
-            self.target_level,
-            self.config.clone(),
-            unused_validator.clone(),
-            self.developer_config.clone(),
-        );
-
-        if let Some(ret) =
-            base_level_compaction_picker.pick_compaction(levels, level_handlers, stats)
+        let l0 = levels.l0.as_ref().unwrap();
+        let overlapping_count = l0
+            .sub_levels
+            .iter()
+            .filter(|level| level.level_type == LevelType::Overlapping as i32)
+            .count();
+        let no_overlap_count = l0
+            .sub_levels
+            .iter()
+            .filter(|level| {
+                level.level_type == LevelType::Nonoverlapping as i32
+                    && level.vnode_partition_count == 0
+            })
+            .count();
+        let partitioned_count = l0
+            .sub_levels
+            .iter()
+            .filter(|level| {
+                level.level_type == LevelType::Nonoverlapping as i32
+                    && level.vnode_partition_count > 0
+            })
+            .count();
+        if (self.config.split_weight_by_vnode == 0 && no_overlap_count > overlapping_count)
+            || (self.config.split_weight_by_vnode > 0
+                && partitioned_count > no_overlap_count
+                && partitioned_count > overlapping_count)
         {
-            return Some(ret);
+            let mut base_level_compaction_picker = LevelCompactionPicker::new_with_validator(
+                self.target_level,
+                self.config.clone(),
+                unused_validator.clone(),
+                self.developer_config.clone(),
+            );
+
+            if let Some(ret) =
+                base_level_compaction_picker.pick_compaction(levels, level_handlers, stats)
+            {
+                return Some(ret);
+            }
         }
+        if self.config.split_weight_by_vnode > 0
```

Comment:
I always think about whether it is right to compact to unconditionally compact l0 to base. This will lead to base-level accumulation. Should we continue to follow the selection rules of the level selector?  What's your opinion?

https://github.com/risingwavelabs/risingwave/issues/14034